src/sciline/pipeline.py

```diff
@@ -750,3 +776,125 @@ def bind_and_call(
+            for c, v in self._subproviders[p].items()
+        )
+
+        providers = groupby(
+            lambda p: p['kind'],
+            (
+                {
+                    'origin': p,
+                    'args': c,
+                    'value': v() if (k := _kind_of_provider(v)) != 'function' else v,
+                    'kind': k,
+                }
+                for p, c, v in chain(
+                    providers_with_parameters, providers_without_parameters
+                )
+            ),
+        )
+
+        param_tables_by_name = groupby(
+            lambda p: p['origin'].label[0].tp, providers['table']
+        )
+        param_tables = []
+        for table_name, table_cells in param_tables_by_name.items():
+            columns = groupby(lambda p: p['origin'].tp, table_cells)
+            index = [p['origin'].label[0].index for p in next(iter(columns.values()))]
+
+            param_tables.append(
+                ParamTable(
+                    table_name,
+                    {k: [v['value'] for v in g] for k, g in columns.items()},
+                    index=index,
+                )
+            )
+
+        def pretty_requirement_name(r: Key | TypeVar) -> str:
+            return '*' if isinstance(r, TypeVar) else f'{_qualname(r)}'
+
+        def pretty_provider_name(p: dict[str, Any]) -> str:
+            name = f'{_qualname(p["origin"])}'
+            if p['args']:
+                name += f'[{",".join(map(pretty_requirement_name, p["args"]))}]'
+            return name
+
+        def provider_source(p: dict[str, Any]) -> str:
+            if p['kind'] != 'function':
+                return ''
+            path = inspect.getsourcefile(p['value'])
+            lineno = inspect.getsourcelines(p['value'])[1]
+            return f'''
+            <details>
+              <summary>{p['value'].__name__}</summary>
+                {path}:{lineno}
+            </details>
+            '''
+
+        def provider_value(p: dict[str, Any]) -> str:
+            if p['kind'] != 'parameter':
+                return ''
+            if hasattr(p['value'], '_repr_html_'):
+                return f'{p["value"]._repr_html_()}'
+            return f'{p["value"]}'
+
+        return f'''
+        <div style="display: flex; column-gap: 1em; align-items: flex-start;">
+        <style>
+          .pipeline-component {{
+              border: 1px solid;
+              padding: 0.5em;
+          }}
+        </style>
+
+        <div class="pipeline-component"
+          {'style="display: none"' if len(providers['function']) == 0 else ''}
+        >
+          <table>
+            <thead>
+              <tr>
+                <th scope="col">Name</th>
+                <th scope="col">Source</th>
+                <th scope="col">Value</th>
+              </tr>
+            </thead>
+            <tbody>
+            {chr(10).join((f"""
+              <tr>
+                <th scope="row">{pretty_provider_name(p)}</th>
```

Comment:
Do we really need to have the names in bold?

Reply:
No not really